src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -27,18 +27,33 @@ static constexpr std::wstring_view PACKAGED_PROFILE_ICON_PATH{ L"ms-appx:///Prof
 static constexpr std::wstring_view PACKAGED_PROFILE_ICON_EXTENSION{ L".png" };
 static constexpr std::wstring_view DEFAULT_LINUX_ICON_GUID{ L"{9acb9455-ca41-5af7-950f-6bca1bc9722f}" };
 
+// make sure this matches defaults.json.
+static constexpr std::wstring_view DEFAULT_WINDOWS_POWERSHELL_GUID{ L"{61c54bbd-c2c6-5271-96e7-009a87ff44bf}" };
+
 // Method Description:
-// - Returns the settings currently in use by the entire Terminal application.
+// - Returns the active application logic singleton
 // Throws:
 // - HR E_INVALIDARG if the app isn't up and running.
-const CascadiaSettings& CascadiaSettings::GetCurrentAppSettings()
+static auto GetAppLogic()
 {
     auto currentXamlApp{ winrt::Windows::UI::Xaml::Application::Current().as<winrt::TerminalApp::App>() };
     THROW_HR_IF_NULL(E_INVALIDARG, currentXamlApp);
 
-    auto appLogic = winrt::get_self<winrt::TerminalApp::implementation::AppLogic>(currentXamlApp.Logic());
-    THROW_HR_IF_NULL(E_INVALIDARG, appLogic);
+    auto appLogicPointer = winrt::get_self<winrt::TerminalApp::implementation::AppLogic>(currentXamlApp.Logic());
+    THROW_HR_IF_NULL(E_INVALIDARG, appLogicPointer);
+
+    winrt::com_ptr<winrt::TerminalApp::implementation::AppLogic> appLogic;
+    appLogic.copy_from(appLogicPointer);
+    return appLogic;

```

Comment:
Wait why are we making _another_ `winrt::com_ptr<winrt::TerminalApp::implementation::AppLogic>` here?

Reply:
so actually get_self returns a raw `implementation::X*` ☹️

Reply:
Honestly i don't know if I care to have this return a `com_ptr`. a bare pointer means it can't be owned, but _the AppLogic should outlive ALL other lifetimes_...